### PR TITLE
Als 1255/move declaration key annotation

### PR DESCRIPTION
**Pull request by agusayerza**
Move declaration keys annotation to declares field

### CODE REVIEW

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/common/BaseSpecParser.scala [hunk 1]
```diff
@@ -26,8 +41,8 @@ trait SpecParserOps {
                                     node: String,
                                     property: String,
                                     ctx: WebApiContext): Unit = {
-    val pattern1 = "\\{[^\\}]*\\{".r
-    val pattern2 = "\\}[^\\{]*\\}".r
+    val pattern1 = "\\{[^}]*\\{".r
+    val pattern2 = "}[^{]*}".r
```
Comment (agusayerza) [inline amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/common/BaseSpecParser.scala:51]: Unrelated code smell

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/common/BaseSpecParser.scala [hunk 2]
```diff
@@ -39,7 +54,7 @@ trait SpecParserOps {
     }
   }
 
-  class ObjectField(target: Target, field: Field)(implicit iv: WebApiContext) extends Function1[YMapEntry, Unit] {
+  class ObjectField(target: Target, field: Field)(implicit iv: WebApiContext) extends (YMapEntry => Unit) {
```
Comment (agusayerza) [inline amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/common/BaseSpecParser.scala:63]: Unrelated code smell

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/AsyncApiDocumentParser.scala [hunk 3]
```diff
@@ -51,15 +51,16 @@ abstract class AsyncApiDocumentParser(root: Root)(implicit val ctx: AsyncWebApiC
...
-    if (declarationKeys.nonEmpty) document.add(DeclarationKeys(declarationKeys))
 
     val api = parseApi(map).add(SourceVendor(ctx.vendor))
     document
       .withEncodes(api)
       .adopted(root.location)
 
     val declarable = ctx.declarations.declarables()
-    if (declarable.nonEmpty) document.withDeclares(declarable)
+    val ann        = Annotations(DeclarationKeys(declarationKeys))
```
Comment (tomsfernandez) [inline amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/AsyncApiDocumentParser.scala]: [OOS] - Not sure if using the ctx to accumulate `DeclarationKeys` is the best approach. Isn't this polluting the already heavily polluted WebApiContext? Besides, DeclarationKeys are only added in DocumentParsers.

File: amf-client/shared/src/test/resources/upanddown/cycle/raml10/extensions/api.raml.expanded.jsonld [hunk 4]
```diff
@@ -2132,6 +2132,7 @@
             ]
           }
         ],
+        "http://a.ml/vocabularies/document#declares": [],
```
Comment (tomsfernandez) [inline amf-client/shared/src/test/resources/upanddown/cycle/raml10/extensions/api.raml.expanded.jsonld]: Hmm why is this "declares" being added?

File: amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/AsyncApiDocumentParser.scala (hunk 3 above)
Comment (agusayerza) [reply amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/async/AsyncApiDocumentParser.scala]: Moved to DeclarationKeysCollector

File: amf-client/shared/src/test/resources/upanddown/cycle/raml10/extensions/api.raml.expanded.jsonld (hunk 4 above)
Comment (agusayerza) [reply amf-client/shared/src/test/resources/upanddown/cycle/raml10/extensions/api.raml.expanded.jsonld]: Fixed

**Review by tomsfernandez — APPROVED** [review]
